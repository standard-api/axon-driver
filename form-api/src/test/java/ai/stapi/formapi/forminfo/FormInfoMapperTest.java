package ai.stapi.formapi.forminfo;

import ai.stapi.test.systemschema.SystemSchemaIntegrationTestCase;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class FormInfoMapperTest extends SystemSchemaIntegrationTestCase {

    @Autowired
    private FormInfoMapper formInfoMapper;

    @Test
    void itCanMapInfoForCreateCommand() {
        var actual = this.formInfoMapper.map("CreateStructureDefinition");
        this.thenObjectApproved(actual);
    }
    
    @Test
    void itCanMapInfoForUpdateCommand() {
        var actual = this.formInfoMapper.map("UpdateStructureDefinition");
        this.thenObjectApproved(actual);
    }

    @Test
    void itCanMapInfoForAddCommandWithoutModificationWithStartId() {
        var actual = this.formInfoMapper.map("AddElementOnStructureDefinitionDifferential");
        this.thenObjectApproved(actual);
    }

    @Test
    void itCanMapInfoForAddCommandWithModificationWithStartId() {
        var actual = this.formInfoMapper.map("AddTypeOnStructureDefinitionDifferentialElement");
        this.thenObjectApproved(actual);
    }
}