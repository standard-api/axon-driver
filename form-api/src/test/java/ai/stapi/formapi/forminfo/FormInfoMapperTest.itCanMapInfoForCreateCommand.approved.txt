{
  "requiresResourceId" : false,
  "resourceType" : "StructureDefinition",
  "possibleTargets" : { }
}